### PR TITLE
Bootstrap flake8 directly instead of relying on flakehell

### PR DESCRIPTION
**Pull request by thejcannon**
# Justification

Although we see a lot of benefits from using `flakehell`, like pyproject.toml support, baseline files, etc... it also has some unfortunate drawbacks. For the time being, I think we should step back from `flakehell` and investigate a solution where we bootstrap Flake8 directly. This also allows the door to be left open for when we're ready to also bootstrap formatting.

# Implementation

- Use `click` and `toml` as support libraries as we're already pulling them in `black` and they are easy to use.
- Repeat the options we want to from `flake8` and then fully manage the `flake8` invocation. This will allow us to stop the need for duplication in the config(s) between the tools we build on top of (E.g. no need to make an `exclude` for `black` and `flake8`, now it's under `ni-python-styleguide`).

# Thoughts for the future

- Add a formatting command which applies fixes for lint errors in addition to running `black`
- Separate out all this business logic into its own library/framework

### CODE REVIEW

**Comment by mshafer-NI** [comment]
```
Run poetry run ni-python-styleguide lint
  poetry run ni-python-styleguide lint
  shell: /bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.6.12/x64
./tests/test_cli/test_bare.py:13:1: BLK100 Black would make changes.
##[error]Process completed with exit code 1.
```

:(

**Comment by DavidCurtiss** [comment]
> Black would make changes.

Yeah, apparently black just released a new version. We hit this in a different project. If your code includes a list that has a trailing comma, e.g. `(1, 2, 3,)` then the new black will think you're asking it to split the list across multiple lines. Maybe there's something else you're hitting here, but that's what it was in the other project.

File: ni_python_styleguide/_cli.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+)
+@click.option(
+    "--extend-exclude",
+    type=str,
+    default="",
+    help="Comma-separated list of files or directories to exclude (in addition to --exclude).",
+)
+@click.version_option()  # @TODO: override the message to include dependency version(s)
+@click.pass_context
+def main(ctx, verbose, quiet, config, exclude, extend_exclude):
```
[the remark this answers is not in the document]
Comment (thejcannon) [reply ni_python_styleguide/_cli.py:91]: Context (from `click.pass_context` :)

File: ni_python_styleguide/_cli.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+def lint(obj, file_or_dir):
+    app = flake8.main.application.Application()
+    filtered_list = lambda iter: list(filter(bool, iter))
+    app.run(
+        filtered_list(
+            [
+                # Ignore all configuration files
+                "--isolated",
+                _qs_or_vs(obj["VERBOSITY"]),
+                # @TODO: pass our config options
```
[the remark this answers is not in the document]
Comment (thejcannon) [reply ni_python_styleguide/_cli.py:110]: oops, good eye!

File: ni_python_styleguide/_cli.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    ctx.ensure_object(dict)
+    ctx.obj["VERBOSITY"] = verbose - quiet
+    ctx.obj["EXCLUDE"] = ",".join(filter(bool, [exclude.strip(","), extend_exclude.strip(",")]))
+
+
+@main.command()
+# @TODO: When we're ready to encourage editor integration, add --diff flag
+@click.argument("file_or_dir", nargs=-1)
+@click.pass_obj
+def lint(obj, file_or_dir):
```
[the remark this answers is not in the document]
Comment (thejcannon) [reply ni_python_styleguide/_cli.py:101]: Pretty much. This makes `lint` a subcommand.

clone my repo and play around with it!

File: ni_python_styleguide/_cli.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+import toml
+
+
+_LINE_LENGTH = 100
+
+
+def _qs_or_vs(verbosity):
+    if verbosity != 0:
+        return f"-{'v' * verbosity if verbosity > 0 else 'q' * abs(verbosity)}"
```
Comment (mshafer-NI) [inline ni_python_styleguide/_cli.py:15]: what if both q and v are used?

**Review by irwand — APPROVED** [review]

File: ni_python_styleguide/_cli.py (hunk 4 above)
Comment (thejcannon) [reply ni_python_styleguide/_cli.py:15]: They duke it out. More prevalent wins.

**Review by mshafer-NI — APPROVED** [review]

**Review by kcuzner — APPROVED** [review]

**Review by neilvana — APPROVED** [review]

**Pull request merged**